identity/src/keypair.rs

```diff
@@ -342,6 +356,32 @@ impl Keypair {
             KeyPairInner::Ecdsa(_) => KeyType::Ecdsa,
         }
     }
+
+    /// Return the secret key of the [`Keypair`] if it has one.
+    #[cfg(any(
+        feature = "ecdsa",
+        feature = "secp256k1",
+        feature = "ed25519",
+        feature = "rsa"
+    ))]
```

Comment:
It would be better to not make this conditional! The `cfg`s inside are enough :)

```suggestion
```

Reply:
I reckon add a line that returns `None` that is cfg'd correctly as the first statement!

You can allow dead-code as wellif that helps.

Reply:
Had to add more `cfg's` to not get
```
  --> identity/src/keypair.rs:355:13
    |
355 |         let secret = match self.keypair {
    |             ^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `[u8]`
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature
   
```
ptal Thomas, thanks